### ISSUE TITLE
Godot 4 is crashing to a white screen immediately on startup

### ISSUE THREAD

**Issue by golddotasksquestions** [godotengine/godot#49656]
### Godot version

4.0.dev.custom_build.12e0f10c7

### System information

Windows 64bit , GeForce GTX 765M

### Issue description

I've been trying to get the Godot 4.0 master Windows artifact builds to run (from https://github.com/godotengine/godot/commits/master) for a couple weeks now, but they always crash immediately and all I see is a white window and crash report.

I have the same issue with Nightly builds from Calinou and iFire.

Also I had no problem running Godot 4 Win artifacts and nightly builds a few months back. I did not keep any of those unfortunately so it is really hard for me to tell exactly when it stopped working.
My drivers are all up to date and my graphics card supports Vulkan
I also tried set various optimus settings (by default I always run Godot with the dedicated GPU), but this made no difference either.

The console window unfortunately is behind the other windows that appear during startup, and quickly disappears, but I've managed to screen record at least some of it and stitch it together in a single png:

![godot4_crashing_on_startup](https://user-images.githubusercontent.com/47016402/122232712-bcff8280-cebb-11eb-874f-642c59892a49.gif) ![Godot4_creash01](https://user-images.githubusercontent.com/47016402/122232796-cab50800-cebb-11eb-9686-8702623cb301.png)

By looking through other crash issues I've stumbled upon [a tip by Calinou](https://github.com/godotengine/godot/issues/48725#issuecomment-841570778) saying to rename The Godot folder in APPDATA, which I tried, but made no difference.

Also in this issue Calinou commented with a technique to copy the error text, so I can also finally fully post it here:

```
Godot Engine v4.0.dev.custom_build.12e0f10c7 - https://godotengine.org
Vulkan API 1.2.162
WARNING: GENERAL - Message Id Number: 0 | Message Id Name: Loader Message
        ReadDataFilesInRegistry: Registry lookup failed to get layer manifest files.
        Objects - 1
                Object[0] - VK_OBJECT_TYPE_INSTANCE, Handle 56088144
     at: VulkanContext::_debug_messenger_callback (drivers\vulkan\vulkan_context.cpp:154)
Using Vulkan Device #0: NVIDIA - GeForce GTX 765M
- Vulkan multiview supported:
  max view count: 32
  max instances: 134217727
- Vulkan subgroup:
  size: 32
  stages: STAGE_VERTEX, STAGE_TESSELLATION_CONTROL, STAGE_TESSELLATION_EVALUATION, STAGE_GEOMETRY, STAGE_FRAGMENT, STAGE_COMPUTE
  supported ops: FEATURE_BASIC, FEATURE_VOTE, FEATURE_ARITHMETIC, FEATURE_BALLOT, FEATURE_SHUFFLE, FEATURE_SHUFFLE_RELATIVE, FEATURE_CLUSTERED, FEATURE_QUAD, FE
ATURE_PARTITIONED_NV
  quad operations in all stages
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
ERROR: Condition "!variants_enabled[p_variant]" is true. Returning: RID()
   at: ShaderRD::version_get_shader (.\servers/rendering/renderer_rd/shader_rd.h:142)
ERROR: Condition "p_shader.is_null()" is true.
   at: PipelineCacheRD::setup (servers\rendering\renderer_rd\pipeline_cache_rd.cpp:67)
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] <couldn't map PC to fn name>
[1] <couldn't map PC to fn name>
[2] <couldn't map PC to fn name>
[3] <couldn't map PC to fn name>
[4] <couldn't map PC to fn name>
[5] <couldn't map PC to fn name>
[6] <couldn't map PC to fn name>
[7] <couldn't map PC to fn name>
[8] vk_optimusGetInstanceProcAddr
[9] vkGetInstanceProcAddr
[10] vk_optimusGetInstanceProcAddr
[11] vk_optimusGetInstanceProcAddr
[12] vkGetInstanceProcAddr
[13] vkGetInstanceProcAddr
[14] vkGetInstanceProcAddr
[15] vkGetInstanceProcAddr
[16] oidnUnmapBuffer
[17] oidnUnmapBuffer
[18] oidnUnmapBuffer
[19] oidnUnmapBuffer
[20] oidnUnmapBuffer
[21] <couldn't map PC to fn name>
[22] oidnUnmapBuffer
[23] <couldn't map PC to fn name>
[24] <couldn't map PC to fn name>
[25] <couldn't map PC to fn name>
[26] <couldn't map PC to fn name>
[27] <couldn't map PC to fn name>
[28] oidnUnmapBuffer
[29] BaseThreadInitThunk
-- END OF BACKTRACE --
```

### Steps to reproduce

1. Download any of the latest artifacts in the master commit or a nightly build.
2. Run the .exe

**Comment by Calinou** [comment]
Which folder did you place the Godot binary into? Is it in a writable location (i.e. *not* Program Files)?

**Comment by golddotasksquestions** [comment]
Thanks for the response @Calinou !

I placed the binary in C:\Users\MyUserName\Gamedev\godot\4.0
So no it's not in  Program Files. The files are locally on my computer. I'm not connected to any network drives.

**Comment by clayjohn** [comment]
It looks like it may be crashing in the GPU driver, but without debug symbols we can't really be sure exactly what is causing the crash. 

Do you know if the source you are getting your builds from has debug builds available as well?

As a note, if the crash is entirely within the NVidia Optimus driver, then debug builds of Godot won't actually help us.

**Comment by golddotasksquestions** [comment]
Thank you for chiming in @clayjohn! Unfortunately I don't know what "debug symbols" are. As I've stated in the description of this issue, I've just always downloaded the Windows editor artifacts and Nightly builds. These are three files I always get: godot.windows.opt.tools.64.exe, godot.windows.opt.tools.64.exp, godot.windows.opt.tools.64.lib

If you could give me instructions on what to do in order to provides you these "debug symbols" I'm really happy to do my best to get you everything you need to get a better picture.

**Comment by Calinou** [comment]
GitHub Actions artifact builds no longer contain debug symbols because it makes binaries much larger, which in turn causes the storage quotas to be exceeded.

**Comment by golddotasksquestions** [comment]
Yeah I think I remember in the beginning when I was testing Godot 4 master builds and it was still working, there were more files and the download was substantially bigger. Is there any possibility for me to download a current master build with all those debug tools?
Could you @Calinou enable debug symbols to be generated for one of your nightly builds? (only once)

**Comment by Calinou** [comment]
> Could you @Calinou enable debug symbols to be generated for one of your nightly builds? (only once)

I'll try to enable debug symbols for editor builds permanently (but not for export templates). Given the current stability of the `master` branch, it's probably better to favor useful crash backtraces over small download sizes.

**Comment by golddotasksquestions** [comment]
So meanwhile I downloaded the latest build from https://hugo.pro/projects/godot-builds/

I was not sure at first this already included the debug symbols because the file size is not really much different and even the zip archive contains just a single exe, last time I downloaded with debug symbols it had a lot more files.

The description says "5 times larger" because of debug symbols, but the download was just two times larger than before so I waited a few days more. Turns out it is actually quite big when unpacked and the extra files may just be included in the exe at least this is how I interpret "debug symbols are kept in the editor binaries".

Anyway. The console printout is a considerably shorter now, I hope is is of use to anyone looking at it nevertheless:

```
Godot Engine v4.0.dev.calinou.60dcc4f39 - https://godotengine.org
Vulkan API 1.2.162
WARNING: GENERAL - Message Id Number: 0 | Message Id Name: Loader Message
        ReadDataFilesInRegistry: Registry lookup failed to get layer manifest files.
        Objects - 1
                Object[0] - VK_OBJECT_TYPE_INSTANCE, Handle 55221360
     at: _debug_messenger_callback (drivers/vulkan/vulkan_context.cpp:154)
Using Vulkan Device #0: NVIDIA - GeForce GTX 765M
- Vulkan multiview supported:
  max view count: 32
  max instances: 134217727
- Vulkan subgroup:
  size: 32
  stages: STAGE_VERTEX, STAGE_TESSELLATION_CONTROL, STAGE_TESSELLATION_EVALUATION, STAGE_GEOMETRY, STAGE_FRAGMENT, STAGE_COMPUTE
  supported ops: FEATURE_BASIC, FEATURE_VOTE, FEATURE_ARITHMETIC, FEATURE_BALLOT, FEATURE_SHUFFLE, FEATURE_SHUFFLE_RELATIVE, 

FEATURE_CLUSTERED, FEATURE_QUAD, FE
ATURE_PARTITIONED_NV
  quad operations in all stages
ERROR: 2 shaders of type SkyShaderRD were never freed
   at: ~ShaderRD (servers/rendering/renderer_rd/shader_rd.cpp:766)
ERROR: 1 shaders of type SdfgiPreprocessShaderRD were never freed
   at: ~ShaderRD (servers/rendering/renderer_rd/shader_rd.cpp:766)
ERROR: 1 shaders of type VoxelGiDebugShaderRD were never freed
   at: ~ShaderRD (servers/rendering/renderer_rd/shader_rd.cpp:766)
ERROR: 1 shaders of type VoxelGiShaderRD were never freed
   at: ~ShaderRD (servers/rendering/renderer_rd/shader_rd.cpp:766)
```

**Comment by Calinou** [comment]
> last time I downloaded with debug symbols it had a lot more files and I remembered it to be a lot bigger in filesize.

Debug symbols can either be a separate file (MSVC `.pdb`, MinGW `.debug`) or embedded within the binary (default behavior with MinGW).

**Comment by golddotasksquestions** [comment]
I'm still getting the same crash on startup, but with slightly different errors:

```
Godot Engine v4.0.dev.calinou.e6a544c16 - https://godotengine.org
Vulkan API 1.2.162
WARNING: GENERAL - Message Id Number: 0 | Message Id Name: Loader Message
        ReadDataFilesInRegistry: Registry lookup failed to get layer manifest files.
        Objects - 1
                Object[0] - VK_OBJECT_TYPE_INSTANCE, Handle 60948608
     at: _debug_messenger_callback (drivers/vulkan/vulkan_context.cpp:154)
Using Vulkan Device #0: NVIDIA - GeForce GTX 765M
- Vulkan multiview supported:
  max view count: 32
  max instances: 134217727
- Vulkan subgroup:
  size: 32
  stages: STAGE_VERTEX, STAGE_TESSELLATION_CONTROL, STAGE_TESSELLATION_EVALUATION, STAGE_GEOMETRY, STAGE_FRAGMENT, STAGE_COMPUTE
  supported ops: FEATURE_BASIC, FEATURE_VOTE, FEATURE_ARITHMETIC, FEATURE_BALLOT, FEATURE_SHUFFLE, FEATURE_SHUFFLE_RELATIVE, 

FEATURE_CLUSTERED, FEATURE_QUAD, FE
ATURE_PARTITIONED_NV
  quad operations in all stages
ERROR: 2 shaders of type SkyShaderRD were never freed
   at: ~ShaderRD (servers/rendering/renderer_rd/shader_rd.cpp:697)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:8313)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:8313)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:8313)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:8313)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:8313)
ERROR: Attempted to free invalid ID: 0
   at: _free_internal (drivers/vulkan/rendering_device_vulkan.cpp:8313)
ERROR: 1 shaders of type SdfgiPreprocessShaderRD were never freed
   at: ~ShaderRD (servers/rendering/renderer_rd/shader_rd.cpp:697)
ERROR: 1 shaders of type VoxelGiDebugShaderRD were never freed
   at: ~ShaderRD (servers/rendering/renderer_rd/shader_rd.cpp:697)
ERROR: 1 shaders of type VoxelGiShaderRD were never freed
   at: ~ShaderRD (servers/rendering/renderer_rd/shader_rd.cpp:697)
```

**Comment by MmAaXx500** [comment]
I think I'm having the same issue. I had played around with it a bit and found the builds with MSVC are fine, with mingw the `debug` is ok and the `release_debug` is crashing.

Obtained the following information from drmingw about the crash:
<details>
<summary>Exception</summary>

```
godot.windows.opt.tools.64.exe caused an Access Violation at location 00007FF73A4C0000 DEP violation at location 00007FF73A4C0000.

AddrPC           Params
00007FF73A4C0000 00000215616FA200 0000003CE61FF5A8 0000003CE61FF5B0
00007FF77C559842 0000000000400000 00007FF77CA33F03 0000000000400000  godot.windows.opt.tools.64.exe!RenderingServerDefault::RenderingServerDefault  [E:/path/core/os/thread.h @ 90]
    88: _FORCE_INLINE_ ID get_id() const { return id; }
    89: // get the ID of the caller thread
>   90: _FORCE_INLINE_ static ID get_caller_id() { return caller_id; }
    91: // get the ID of the main thread
    92: _FORCE_INLINE_ static ID get_main_id() { return main_thread_id; }
00007FF77A4DE90B 0000021561579CC0 0000021560C1F690 00000215000000FF  godot.windows.opt.tools.64.exe!Main::setup2  [E:/path/main/main.cpp @ 1590]
  1588: /* Initialize Rendering Server */
  1589: 
> 1590: rendering_server = memnew(RenderingServerDefault(OS::get_singleton()->get_render_thread_mode() == OS::RENDER_SEPARATE_THREAD));
  1591: 
  1592: rendering_server->init();
00007FF77A4E479B 0000021560C20C30 0000000000000001 000002155F450860  godot.windows.opt.tools.64.exe!Main::setup  [E:/path/main/main.cpp @ 1388]
  1386: 
  1387: if (p_second_phase) {
> 1388: return setup2();
  1389: }
  1390: 
00007FF77A4C18A7 000002155F153296 0000003CE61FFD4C 0000000000000001  godot.windows.opt.tools.64.exe!widechar_main  [E:/path/platform/windows/godot_windows.cpp @ 151]
   149: TEST_MAIN_PARAM_OVERRIDE(argc, argv_utf8)
   150: 
>  151: Error err = Main::setup(argv_utf8[0], argc - 1, &argv_utf8[1]);
   152: 
   153: if (err != OK) {
00007FF77A4C19A0 0000000000000000 0000000000000047 00007FF7811164D8  godot.windows.opt.tools.64.exe!_main  [E:/path/platform/windows/godot_windows.cpp @ 185]
   183: }
   184: 
>  185: result = widechar_main(argc, wc_argv);
   186: 
   187: LocalFree(wc_argv);
00007FF77A4C13B1 0000000000000000 0000000000000000 0000000000000000  godot.windows.opt.tools.64.exe!__tmainCRTStartup  [C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/crtexe.c @ 321]
00007FF77A4C14E6 0000000000000000 0000000000000000 0000000000000000  godot.windows.opt.tools.64.exe!mainCRTStartup  [C:/_/M/mingw-w64-crt-git/src/mingw-w64/mingw-w64-crt/crt/crtexe.c @ 202]
00007FF8F8997034 0000000000000000 0000000000000000 0000000000000000  KERNEL32.DLL!BaseThreadInitThunk
00007FF8FA1C2651 0000000000000000 0000000000000000 0000000000000000  ntdll.dll!RtlUserThreadStart
```

</details>

**Comment by MmAaXx500** [comment]
For me 99fe462452be44efa618e83ad9bbecd722ae6ecd introduced the bug.

**Comment by akien-mga** [comment]
Can you test the build linked here? https://github.com/godotengine/godot/issues/50970#issuecomment-889308948

**Comment by akien-mga** [comment]
@MmAaXx500 Your issue is likely different from @golddotasksquestions who is using MSVC builds and has a crash in the Vulkan driver. Please open a new issue with details and your findings.

**Comment by MmAaXx500** [comment]
@akien-mga Opened one here: #51070

**Comment by golddotasksquestions** [comment]
I'm super happy, because for the first day in months I was able to start Godot4. No instant crash to white screen any more! I was already afraid I would have to buy a new computer just to run Godot4 ...

I'm using the Calinou's nightly build Godot v4.0.dev.calinou [036b7a098]

It currently takes a _lot_ longer to start up, or load a project and generally do anything in the Editor, but I guess this is to be expected in a prealpha. I just happy I can test things now and look around! 

Closing this issue as it seems to be fixed!
